src/Utilities/OperatingSystemFilterHelpers.js

```diff
@@ -4,24 +4,44 @@ import mapValues from 'lodash/mapValues';
 import { coerce, compare, rcompare } from 'semver';
 import { OS_CHIP } from './constants';
 
-export const updateGroupSelectionIdentifier = (selection, major) =>
-  // if every minor version is selected, then mark the group as selected
+export const updateGroupSelectionIdentifier = (selection, groupLabel, major) =>
   set(
     selection,
-    [major, major],
-    Object.values({ ...selection[major] })
+    [groupLabel, groupLabel],
+    Object.values({ ...selection[groupLabel] })
       .filter((v) => v !== major)
       .every(Boolean)
   );
 
-/** Takes an array of string versions `value` and returns an object in the format
+const isVersionSelected = (selectedVersion, osVersion) => {
+  for (let i = 0; i < selectedVersion.length; i++) {
+    if (
+      selectedVersion[i].value === osVersion.value &&
+      selectedVersion[i].osName === osVersion.osName
+    ) {
+      return true;
+    } else {
+      continue;
+    }
+  }
+
+  return false;
```

Comment:
This could be a `.some`[0] like:

```js

const isVersionSelected = (selectedVersions, osVersion) =>
  selectedVersions.some(({value, osName}) =>  value === osVersion.value && osName === osVersion.osName)

```

[0] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some